src/silx/gui/plot/ImageStack.py

```diff
@@ -581,10 +498,12 @@ def setCurrentUrl(self, url: typing.Union[DataUrl, str]) -> None:
             self._plot.clear()
         else:
             if self._current_url.path() in self._urlData:
-                self._plot.setData(self._urlData[url.path()])
+                self._waitingOverlay.setVisible(False)
+                self._plot.addImage(self._urlData[url.path()], resetzoom=self._autoResetZoom)
             else:
+                self._plot.clear()
```

Comment:
Question: Do you want to clear the previous image while loading?

Reply:
With the current implementation of the 'WaiterOverlay' this is not mandatory to keep it.
Now the question is more: should we keep the data in the `Plot` until it is replaced... it will be there but not displayed.

I would say we can let it so we free resources ASAP but this is not a strong opinion. If you want we can remote it as well (would do one less line to maintain ;)).

Reply:
Better for user understanding to clear it. Increase readability.
We could avoid it by defining Waiter-overlay style but lets keep it simple.